calendarsubscription/calendarsubscription.go

```diff
@@ -0,0 +1,39 @@
+package calendarsubscription
+
+import (
+	"time"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/target/goalert/validation/validate"
+)
+
+type CalendarSubscription struct {
+	ID         string
+	Name       string
+	UserID     string
+	ScheduleID string
+	LastAccess time.Time
+	Disabled   bool
+
+	// Config provides necessary parameters CalendarSubscription Config (i.e. ReminderMinutes)
+	Config struct {
+		ReminderMinutes []int
+	}
+}
+
+func (cs CalendarSubscription) Normalize() (*CalendarSubscription, error) {
+	if cs.ID == "" {
+		cs.ID = uuid.NewV4().String()
+	}
+
+	err := validate.Many(
+		validate.IDName("Name", cs.Name),
+		validate.UUID("ID", cs.ID),
+		validate.UUID("UserID", cs.UserID),
```

Comment:
We should probably add validation here for returning error if reminderMinutes is negative(example: -1). Currently it accepts all int values.

Reply:
I think negative is OK, it would mean a user wants an extra notification say 5 minutes after their shift starts. As long as the input doesn't cause us to output something wrong (i.e. as long as we can generate a valid iCal/VALARM entry) it's probably okay.

Reply:
Having said that, we def. should validate the length of the array at least before serializing and sending it to the DB.

Reply:
That sounds good to me. Should length of array be restricted to a specific limit?

Reply:
Yeah I think so. Maybe 15? I think on the UI we were thinking of making sure 3 looks good, and max of 5. API could be a little more flexible though -- the goal is for nobody to ever know there is a limit during normal usage, but keep it bounded in the case of mistakes or "curiosity"